core-container/entry.sh

```diff
@@ -43,56 +33,78 @@ function set_provider_adapters() {
     rm kcpsecret
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl delete spaceproviderdesc $PROVIDER_NAME ; then
         echo "Nothing to delete."
     fi
     kubectl create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+}
  
-        echo "Delete the kubeflex provider object if it already exists, and then create it."
-        if ! kubectl delete spaceproviderdesc pkflex ; then
-            echo "Nothing to delete."
-        fi
-        kubectl create -f - <<EOF
+function create_kubeflex_provider() {
+    echo "Waiting for the kubeflex provider to be ready... this may take a while."
+    (
+        until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
```

Comment:
It would be better to do this with real programming instead of text processing. How about doing it like in https://docs.kubestellar.io/main/Coding%20Milestones/PoC2023q1/environments/dev-env/#create-an-nginx-ingress-controller-with-ssl-passthrough ? That also covers the looping.

Reply:
https://github.com/kubestellar/kubestellar/issues/1332